pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/endpoints/objects/internal/IncludeQueryParam.kt

```diff
@@ -1,46 +1,55 @@
 package com.pubnub.internal.endpoints.objects.internal
 
-import com.pubnub.api.models.consumer.objects.member.PNUUIDDetailsLevel
-import com.pubnub.api.models.consumer.objects.membership.PNChannelDetailsLevel
-
 data class IncludeQueryParam(
+    // common
     private val includeCustom: Boolean = false,
-    private val includeChannelDetails: PNChannelDetailsLevel? = null,
-    private val includeUUIDDetails: PNUUIDDetailsLevel? = null,
     private val includeType: Boolean = true,
     private val includeStatus: Boolean = true,
+    // channel(membership) related
+    private val includeChannel: Boolean = false,
+    private val includeChannelCustom: Boolean = false,
+    private val includeChannelStatus: Boolean = false,
     private val includeChannelType: Boolean = false,
-    private val includeUuidType: Boolean = false
+    // user(member) related
+    private val includeUser: Boolean = false,
+    private val includeUserCustom: Boolean = false,
+    private val includeUserStatus: Boolean = false,
+    private val includeUserType: Boolean = false,
 ) {
+    internal companion object {
+        const val QUERY_PARAM_CUSTOM = "custom"
+        const val QUERY_PARAM_TYPE = "type"
+        const val QUERY_PARAM_STATUS = "status"
+        const val QUERY_PARAM_CHANNEL = "channel"
+        const val QUERY_PARAM_CHANNEL_CUSTOM = "channel.custom"
+        const val QUERY_PARAM_CHANNEL_STATUS = "channel.status"
+        const val QUERY_PARAM_CHANNEL_TYPE = "channel.type"
+        const val QUERY_PARAM_UUID = "uuid"
+        const val QUERY_PARAM_UUID_CUSTOM = "uuid.custom"
+        const val QUERY_PARAM_UUID_STATUS = "uuid.status"
+        const val QUERY_PARAM_UUID_TYPE = "uuid.type"
+        const val QUERY_PARAM_INCLUDE = "include"
```

Comment:
this should have a different name than the others, as it's not one of the values, it's the key (it can be misleading now)

maybe just `QUERY_KEY_INCLUDE`

Reply:
Makes sense. I will change it.